kymatio/scattering2d/core/scattering2d.py

```diff
@@ -0,0 +1,56 @@
+# Authors: Edouard Oyallon
+# Scientific Ancestry: Edouard Oyallon, Laurent Sifre, Joan Bruna
+
+
+__all__ = ['scattering2d']
+
+def scattering2d(x, pad, unpad, backend, J, L, phi, psi, max_order):
+    subsample_fourier, modulus, fft, cdgmm, finalize = backend.subsample_fourier, backend.modulus,\
+                                                         backend.fft, backend.cdgmm, backend.finalize
+    order0_size = 1
+    order1_size = L * J
+    order2_size = L ** 2 * J * (J - 1) // 2
+    output_size = order0_size + order1_size
+
+    if max_order == 2:
+        output_size += order2_size
+
+    out_S_0, out_S_1, out_S_2 = [], [], []
+    U_r = pad(x)
+    U_0_c = fft(U_r, 'C2C')
+
+    # First low pass filter
+    U_1_c = subsample_fourier(cdgmm(U_0_c, phi[0]), k=2**J)
```

Comment:
A lot of these subsample fourier have a cdgmm going on inside of them. Do we want to separate all instances of two? More lines of code, but it would be a lot more clear.

Reply:
good point, I agree